Fail to build Histoire-vendors with npm 9.6.2

### Describe the bug

```
PS D:\Library\Projects\histoire\packages\histoire-vendors> pnpm run build

> @histoire/vendors@0.15.8 build D:\Library\Projects\histoire\packages\histoire-vendors
> rimraf dist && rollup -c rollup.config.mjs


src/client/b-floating-vue.ts, src/client/b-iconify.ts, src/client/b-pinia.ts, src/client/b-scroll.ts, src/client/b-vue-router.ts, src/client/b-vue-use.ts, src/client/b-vue.ts → dist/client...
Error: Command failed with exit code 1: npm install --prefer-offline
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: histoire-vendors-temp@0.0.0
npm ERR! Found: vue@undefined
npm ERR! node_modules/vue
npm ERR!   vue@"3.2.47" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer vue@"^3.2.0" from floating-vue@2.0.0-beta.20
npm ERR! node_modules/floating-vue
npm ERR!   floating-vue@"2.0.0-beta.20" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! C:\Users\hugoa\AppData\Local\npm-cache\_logs\2023-03-30T16_36_39_215Z-eresolve-report.txt
```

```
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\hugoa\AppData\Local\npm-cache\_logs\2023-03-30T16_36_39_215Z-debug-0.log
    at makeError (file:///D:/Library/Projects/histoire/node_modules/.pnpm/execa@6.1.0/node_modules/execa/lib/error.js:59:11)
    at execaSync (file:///D:/Library/Projects/histoire/node_modules/.pnpm/execa@6.1.0/node_modules/execa/index.js:193:17)
    at Object.closeBundle (file:///D:/Library/Projects/histoire/packages/histoire-vendors/rollup.config.mjs:82:13)
    at D:\Library\Projects\histoire\node_modules\.pnpm\rollup@3.14.0\node_modules\rollup\dist\shared\rollup.js:24410:40
    at async Promise.all (index 0)
    at PluginDriver.hookParallel (D:\Library\Projects\histoire\node_modules\.pnpm\rollup@3.14.0\node_modules\rollup\dist\shared\rollup.js:24338:9)
    at Object.close (D:\Library\Projects\histoire\node_modules\.pnpm\rollup@3.14.0\node_modules\rollup\dist\shared\rollup.js:25528:13)
    at build (D:\Library\Projects\histoire\node_modules\.pnpm\rollup@3.14.0\node_modules\rollup\dist\bin\rollup:1555:5)
    at runRollup (D:\Library\Projects\histoire\node_modules\.pnpm\rollup@3.14.0\node_modules\rollup\dist\bin\rollup:1660:21) {
  shortMessage: 'Command failed with exit code 1: npm install --prefer-offline',
  command: 'npm install --prefer-offline',
  escapedCommand: 'npm install --prefer-offline',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: 'npm ERR! code ERESOLVE\n' +
    'npm ERR! ERESOLVE unable to resolve dependency tree\n' +
    'npm ERR! \n' +
    'npm ERR! While resolving: histoire-vendors-temp@0.0.0\n' +
    'npm ERR! Found: vue@undefined\n' +
    'npm ERR! node_modules/vue\n' +
    'npm ERR!   vue@"3.2.47" from the root project\n' +
    'npm ERR! \n' +
    'npm ERR! Could not resolve dependency:\n' +
    'npm ERR! peer vue@"^3.2.0" from floating-vue@2.0.0-beta.20\n' +
    'npm ERR! node_modules/floating-vue\n' +
    'npm ERR!   floating-vue@"2.0.0-beta.20" from the root project\n' +
    'npm ERR! \n' +
    'npm ERR! Fix the upstream dependency conflict, or retry\n' +
    'npm ERR! this command with --force or --legacy-peer-deps\n' +
    'npm ERR! to accept an incorrect (and potentially broken) dependency resolution.\n' +
    'npm ERR! \n' +
    'npm ERR! \n' +
    'npm ERR! For a full report see:\n' +
    'npm ERR! C:\\Users\\hugoa\\AppData\\Local\\npm-cache\\_logs\\2023-03-30T16_36_39_215Z-eresolve-report.txt\n' +
    '\n' +
    'npm ERR! A complete log of this run can be found in:\n' +
    'npm ERR!     C:\\Users\\hugoa\\AppData\\Local\\npm-cache\\_logs\\2023-03-30T16_36_39_215Z-debug-0.log',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
created dist/client in 11.1s
```

### Reproduction

```
pnpm run build
```

### System Info

```shell
n/a
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/histoire-dev/histoire/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/histoire-dev/histoire/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://histoire.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/histoire-dev/histoire/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/histoire-dev/histoire/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/histoire-vendors/rollup.config.mjs
+++ b/packages/histoire-vendors/rollup.config.mjs
@@ -79,7 +79,7 @@ export * from '${filepath}'\n`
           {
             const tempPkgFile = path.resolve(tempDir, 'package.json')
             fs.writeJsonSync(tempPkgFile, tempPkg)
-            execaSync('npm', ['install', '--prefer-offline'], { cwd: tempDir })
+            execaSync('npm', ['install', '--prefer-offline --legacy-peer-deps'], { cwd: tempDir })
             const dtsFiles = globbySync(['**/*.d.ts', '**/package.json'], {
               cwd: path.join(tempDir, 'node_modules'),
               dot: true,

